Add note for freshness stating the requirement to have both count and… (#7460)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/state-aware-setup.md

To do this, she uses the model `freshness` config. This config helps state-aware orchestration decide _when_ a model should be rebuilt. 

> Note that for every `freshness` config, it is required to either set values for both `count` and `period`, or set `freshness: null`. This requirement applies to all `freshness` types: `freshness.warn_after`, `freshness.error_after`, and `freshness.build_after`.

Reviewer: ```suggestion
Note that for every `freshness` config, you're required to either set values for both `count` and `period`, or set `freshness: null`. This requirement applies to all `freshness` types: `freshness.warn_after`, `freshness.error_after`, and `freshness.build_after`.
```
